Fix custom taxonomy creation steps and conf (#1723)
ezsystems/developer-documentation: docs/content_management/taxonomy/taxonomy.md

## Customize taxonomy structure

You can create other taxonomies than the one predefined in the system, for example a Content category.

> To do it, first, create a new container to store the new taxonomy's items, for example a `folder` named Content categories.

Author: A nice trick here is to open the camouflaged Taxonomy folder (admin/)view/content/56/full/1/57 and to create this container along side Tags.

I don't know if it's a good idea to document this nor how it could be documented.

Any advice?

> To do it, first, create a new container to store the new taxonomy's items, for example a `folder` named Content categories.

Reviewer: ```suggestion
To do it, first, create a new container to store the new taxonomy's items, for example a folder named "Content categories".
```
Re: trick, I guess we need to make sure this is intended way to use taxonomy, ping @juskora

Reviewer 2: I'm not sure, I think better idea would be to ask Maciej Kobus?

Next, in `config/packages/ibexa_taxonomy.yaml` add the following configuration:

``` yaml
[[= include_file('code_samples/taxonomy/config/packages/ibexa_taxonomy.yaml')=]]
```

> Replace `<content_categories_remote_id>` with the new container's location remote ID.

Reviewer: ```suggestion
Replace `<content_categories_remote_id>` with the new container's Location remote ID.
```

Translate the configuration identifier in the `ibexa_taxonomy` domain by, for example, creating a `translations/ibexa_taxonomy.en.yaml` file containing the following:
```yaml
taxonomy.content_categories: 'Content categories'
```

> Then, create a Content Type with `content_category` identifier and include the following Field Types:

Reviewer: ```suggestion
Then, create a Content Type with `content_category` identifier and include the following Field definitions:
```

> * `name_field` of `ezstring` type and required

Author: ```suggestion
* `name_field` of `ezstring` type and required, also used as the name pattern `<name_field>`
```
Hopping it is not too confusing.

Any advice?

> * `name_field` of `ezstring` type and required

Reviewer: How about:
```suggestion
* `name_field` of `ezstring` type and required. Use this Field, as `<name_field>`, for content name pattern.
```

* `category_identifier` of `ezstring` type and required

> * `parent_category` of `ibexa_taxonomy_entry` type and not required, in its Taxonomy drop-down menu, select Content categories (or `taxonomy.content_categories` if no translation has been provided).

Reviewer: ```suggestion
* `parent_category` of `ibexa_taxonomy_entry` type and not required. In its Taxonomy drop-down menu, select Content categories (or `taxonomy.content_categories` if no translation has been provided).
```

Finish taxonomy set-up by creating a new Content category named Root with identifier `content_categories_root` under the previously created container folder named Content categories.

> To use this new taxonomy, add an `ibexa_taxonomy_entry_assignement` field to a Content Type and select Content categories (or `taxonomy.content_categories`) in its Taxonomy drop-down setting.

Reviewer: ```suggestion
To use this new taxonomy, add an `ibexa_taxonomy_entry_assignement` Field to a Content Type and select Content categories (or `taxonomy.content_categories`) in its Taxonomy drop-down setting.
```

You can create other taxonomies than the one predefined in the system, for example a Content category.

To do it, first, create a new container to store the new taxonomy's items, for example a folder named "Content categories".

Next, in `config/packages/ibexa_taxonomy.yaml` add the following configuration:

``` yaml
[[= include_file('code_samples/taxonomy/config/packages/ibexa_taxonomy.yaml')=]]
```

Replace `<content_categories_remote_id>` with the new container's Location remote ID.

Translate the configuration identifier in the `ibexa_taxonomy` domain by, for example, creating a `translations/ibexa_taxonomy.en.yaml` file containing the following:
```yaml
taxonomy.content_categories: 'Content categories'
```

Then, create a Content Type with `content_category` identifier and include the following Field definitions:

* `name` of `ezstring` type and required. Use this Field, as `<name>`, for content name pattern.
* `category_identifier` of `ezstring` type and required.
* `parent_category` of `ibexa_taxonomy_entry` type and not required. In its Taxonomy drop-down menu, select Content categories (or `taxonomy.content_categories` if no translation has been provided).

> Finish taxonomy set-up by creating a new Content category named Root with identifier `content_categories_root` under the previously created container folder named Content categories.

Reviewer: ```suggestion
Finish taxonomy setup by creating a new Content category named Root with identifier `content_categories_root` under the previously created container folder named Content categories.
```
